content/topics/dotenv-module/_index.md

```diff
@@ -0,0 +1,208 @@
+
+```
+
+## How do I use dotenv?
+
+- Create project directory and name it whatever you want.
+- cd into the directory you created and run
+```
+npm init -y
+npm install --save dotenv
+
+``` 
+- Create a .env file and add your settings.
+
+```
+//.env
+
+BASE_URL=https://umuzi.org
+API_KEY=000FBAA4558FF565851A3E104017AB1A
+SERVER_PORT=3000
+USER=umuziadmin
+OBJ={key1:"value1", key2:"value2",key3:"value3",key4:"value4"}
+BOOL=true
+NUM=123456
+STRING= Dumela Lefatshe
+NULL=null
+UNDEFINED=undefined
+ARRAY=[1,2,3,4,5,6]
+
+```
+
+#### &nbsp; Usage
+
+- Create a .js file and start using the module
+```
+//index.js
+
+require('dotenv').config();
+
+const url = process.env.BASE_URL;
+const apiKey = process.env.API_KEY;
+const port = process.env.SERVER_PORT;
+const user = process.env.USER;
+const obj = process.env.OBJ;
+const boolean = process.env.BOOL;
+const number = process.env.NUM;
+const string = process.env.STRING;
+const null1 = process.env.NULL;
+const undefined1 = process.env.UNDEFINED;
+const array = process.env.ARRAY;
+
+console.log(url);
+console.log(apiKey);
+console.log(port);
+console.log(user); //lol, this one seems to return the name of pc owners.
+console.log(obj);
+console.log(number);
+console.log(string);
+console.log(boolean);
+console.log(null1 == null ? true : false);
+console.log(array);
+
+```
+
+What if you need this environment variables in multiple places? If you do reference the variables everywhere that you need them it could make refactoring and maintenance more difficult than if they are in one place. 
+
+#### &nbsp; You can use a Javascript config file to get around this.
+
+```
+//config.js
+
+require('dotenv').config();
+
+module.exports = {
+    url : process.env.BASE_URL,
+    apiKey : process.env.API_KEY,
+    port : process.env.SERVER_PORT,
+    user : process.env.USER,
+    obj : process.env.OBJ,
+    boolean : process.env.BOOL,
+    number : process.env.NUM,
+    string : process.env.STRING,
+    null1 : process.env.NULL,
+    undefined1 : process.env.UNDEFINED,
+    array : process.env.ARRAY
+};
+
+```
+
+Now you can import this configs from any place in your code.
+
+```
+//index.js, doSomething.js, anotherJsFile.js, etc
+
+const configs = require('/path/to/js/config/file');
+
+console.log(configs.url);
+console.log(configs.apiKey);
+console.log(configs.user);
+console.log(configs.obj);
+
+```
+##### What’s the value in this technique?
+
+- it is easy
+- clarity on how all environment variables are being mapped
+- you can rename variables to more readable properties
+- you can add other configuration properties from non-environment variables
+
+## &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; OR
+
+```
+//config.js
+
+const dotenv = require('dotenv');
+const result = dotenv.config();
+
+if (result.error) {
+  throw result.error;
+}
+
+const { parsed: envs } = result; 
+
+module.exports = envs;
```

Comment:
I don't really like this approach. Reason being:
- explicit is better than implicit
- the required environmental variables should be documented somewhere. It might as well be documented as code